htdocs/editlog.php

```diff
@@ -15,527 +15,432 @@
-                            strtolower(
-                                $locale
-                            ) . '"></script>' . "\n";
-                        tpl_set_var('descMode', 3);
-                    }
-                }
-                $headers .= '<script language="javascript" type="text/javascript" src="' . editorJsPath() . '"></script>' . "\n";
-                tpl_set_var('htmlheaders', $headers);
-
-                if ($use_log_pw == true && $log_pw != '') {
-                    if (!$pw_ok && isset($_POST['submitform'])) {
-                        tpl_set_var('log_pw_field', $log_pw_field_pw_not_ok);
-                    } else {
-                        tpl_set_var('log_pw_field', $log_pw_field);
-                    }
-                } else {
-                    tpl_set_var('log_pw_field', '');
-                }
-
-                // build smilies
-                $smilies = '';
-                if ($descMode != 3) {
-                    $countShow = count($smiley['show']);
-                    for ($i = 0; $i < $countShow; $i++) {
-                        if ($smiley['show'][$i] == '1') {
-                            $tmp_smiley = $smiley_link;
-                            $tmp_smiley = mb_ereg_replace('{smiley_image}', $smiley['image'][$i], $tmp_smiley);
-                            $tmp_smiley = mb_ereg_replace('{smiley_symbol}', $smiley['text'][$i], $tmp_smiley);
-                            $smilies = $smilies . '&nbsp;' .
-                                mb_ereg_replace(
-                                    '{smiley_file}',
-                                    $smiley['file'][$i],
-                                    $tmp_smiley
-                                ) . '&nbsp;';
-                        }
-                    }
-                    tpl_set_var('smileypath', $opt['template']['smiley']);
-                }
-                tpl_set_var('smilies', $smilies);
-            }
+        $oldDescMode = $descMode;
+    }
+} else {
+    if ($log_record['text_html'] == 1) {
+        if ($log_record['text_htmledit'] == 1) {
+            $descMode = 3;
+        } else {
+            $descMode = 2;
         }
+    } else {
+        $descMode = 1;
     }
+
+    $oldDescMode = $descMode;
 }
 
-tpl_set_var('scrollposx', isset($_REQUEST['scrollposx']) ? $_REQUEST['scrollposx'] + 0 : 0);
-tpl_set_var('scrollposy', isset($_REQUEST['scrollposy']) ? $_REQUEST['scrollposy'] + 0 : 0);
-
-//make the template and send it out
-tpl_BuildTemplate();
-
-/**
- * @param $cacheId
- * @param $logId
- * @return bool
- */
-function is_latest_log($cacheId, $logId)
-{
-    $latestLogId = sqlValue(
-        "
-        SELECT `id` FROM `cache_logs`
-        WHERE `cache_id`='" . sql_escape($cacheId) . "'
-        ORDER BY `order_date` DESC, `date_created` DESC, `id` DESC
-        LIMIT 1",
-        0,
-        $cacheId
+// fuer alte Versionen von OCProp
+if (isset($_POST['submit']) && !isset($_POST['version2'])) {
```

Comment:
Wenn das If Statement nur für OcProp da ist - bitte entfernen.

Reply:
Das if-Statement verhindert, dass alte Ocprop-Versionen fehlerhafte Logs hochladen. Sicher, dass das raus soll?

Reply:
Ja - ocprop wird blockiert da nur eine neue Version den Server nicht mit Anfragen überfluten würde. Wir mussten daher alle alten Versionen sperren. Den Patch hab ich schon glaube vor einem drei viertel Jahr eingereicht. Leider ist nichts passiert. 

Daher baue ich aktuell ein Möglichkeit in der man die founds.gpx von geocaching.com hochladen und damit seine Funde abgleichen kann. Das gleiche baue ich auch für die Caches.

Reply:
> ocprop wird blockiert da nur eine neue Version den Server nicht mit Anfragen überfluten würde.

Fällt mir eben noch ein: Anfragefluten gab es bei Ocprop immer. Also wenn zwei GC-Logger mit tausenden von Funden zufällig gleichzeitig ihren (z.B.) quartalsweisen Abgleich machen, dann ist der Server die nächsten 1-2 Stunden hauptsächlich mit Ocprop beschäftigt. Konnte man aber mit leben, die Seite blieb bedienbar.

Nur einmal wure es echt kritisch, weil jemand Ocprop täglich hat laufen lassen; den haben wir dann angemailt und danach war gut.